classes/class-dintero-checkout-callback.php

```diff
@@ -282,21 +282,18 @@ public static function is_localhost() {
 	 * @return int
 	 */
 	public function get_order_id_from_reference( $merchant_reference ) {
-		$query_args = array(
-			'fields'      => 'ids',
-			'post_type'   => wc_get_order_types(),
-			'post_status' => array_keys( wc_get_order_statuses() ),
-			'meta_key'    => '_dintero_merchant_reference', // phpcs:ignore WordPress.DB.SlowDBQuery -- Slow DB Query is ok here, we need to limit to our meta key.
-			'meta_value'  => $merchant_reference, // phpcs:ignore WordPress.DB.SlowDBQuery -- Slow DB Query is ok here, we need to limit to our meta key.
+		$orders = wc_get_orders(
+			array(
+				'meta_key'   => '_dintero_merchant_reference',
+				'meta_value' => $merchant_reference,
+				'limit'      => 1,
+				'orderby'    => 'date',
+				'order'      => 'DESC',
+			)
 		);
 
-		$order_ids = get_posts( $query_args );
-
-		if ( empty( $order_ids ) ) {
-			return null;
-		}
-
-		return $order_ids[0];
+		$order = reset( $orders );
+		return $order->get_id() ?? 0;
```

Comment:
Add in some sort of validation that the order we received actually has the value we asked for as the metadata. Since the WordPress query can be an issue with that it will just return something if the query is incorrect, this will let us safeguard against potential issues.